pilot/pkg/model/push_context.go

```diff
@@ -768,6 +771,52 @@ func (ps *PushContext) GatewayServices(proxy *Proxy) []*Service {
 	return gwSvcs
 }
 
+// add services from MeshConfig.ExtensionProviders
+// TODO: include cluster from EnvoyFilter such as global ratelimit [demo](https://istio.io/latest/docs/tasks/policy-enforcement/rate-limit/#global-rate-limit)
+func (ps *PushContext) getHostsFromMeshConfig(proxy *Proxy) sets.Set {
+	hostsFromMeshConfig, applicableProviders := sets.NewSet(), sets.NewSet()
+
+	for _, p := range ps.Telemetry.AccessLogging(proxy).Providers {
```

Comment:
My initial thought was we would just get all the `ExtensionProviders`, not only the applicable ones. Since this is missing some other uses, like External Authz?

If not, it seems a bit wasteful to compute the full config (expensive) just to get the providers?

I think this is missing `metrics` as well?

Reply:
> My initial thought was we would just get all the ExtensionProviders, not only the applicable ones. Since this is missing some other uses, like External Authz?

that's what I do in the first version, then I take a look at `Metric` `Trace` and `Logging`, but I forgot `External Authz`. 
you're mostly right, it would be better for just get all the `ExtensionProviders`.

> I think this is missing metrics as well?

IIRC, there're no address on `prometheus` and `stackdriver`.

Reply:
updated